File: Semaine 2 Projet/WayGo/app/src/main/java/com/teammirado/waygo/data/model/Booking.kt
package com.teammirado.waygo.data.model

import com.google.gson.annotations.SerializedName

data class Booking(
    val id: Int,
    @SerializedName("trip_id") val tripId: Int,
    @SerializedName("user_id") val userId: Int,
    val status: String // e.g., "pending", "confirmed", "cancelled"
)

File: backend laravel/app/Models/Booking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Booking extends Model {
    use HasFactory;

    // 🛠️ CORRECTION : Autorise 'seats' et 'seats_booked' pour s'aligner parfaitement avec le contrôleur
    protected $fillable = [
        'trip_id',
        'passenger_id',
        'seats',        // Utilisé dans le contrôleur Laravel
        'seats_booked', // Alternative historique
        'status',       // 'pending', 'confirmed', 'rejected', 'cancelled'
    ];

    /**
     * Une réservation appartient à un trajet spécifique.
     */
    public function trip() {
        return $this->belongsTo(Trip::class);
    }

    /**
     * Une réservation appartient à un passager (un utilisateur).
     * Configuration validée : fait la passerelle parfaite avec le contrôleur !
     */
    public function passenger() {
        return $this->belongsTo(User::class, 'passenger_id');
    }
}